### PR TITLE
feat: Add non-interactive state to card header

### PR DESCRIPTION
**Pull request by JKMarkowski**
## Related Issue
Closes https://github.com/SAP/fundamental-styles/issues/2097

## Description
There is added new `fd-card__header--non-interactive` class modifier. And description of its usage.

#### Please check whether the PR fulfills the following requirements

1. The output matches the design specs
- [x] hover state of the element follow design spec
- [x] focus state of the element follow design spec
- [x] active state of the element follow design spec
- [x] selected state of the element follow design spec
- [x] selected hover state of the element follow design spec
- [x] pressed state of the element follow design spec
2. The code follows fundamental-styles code standards and style
- [x] BEM naming convention is used
- [x] Mixins are used for repeatable code (`fd-rtl`, `fd-ellipsis`, `fd-flex`, `fd-selected`, `fd-focus`, ect.)
- [x] A11y support - keyboard support, screenreader support, proper ARIA attributes, etc.
- [x] `fd-reset()` mixin is applied to all element
- [x] Variables are used, if some value is used more than twice.
- [x] Checked if current components can be reused, instead of having new code.
3. Testing
- [x] Verified all styles in IE11
- [x] Updated tests
- [x] last commit message should have `[ci visual]` so it can trigger chromatic visual regression (e.g. `test: run chromatic visual regression [ci visual]`)
4. Documentation
- [x] Storybook documentation has been created/updated
- [x] Breaking Changes [wiki](https://github.com/SAP/fundamental-styles/wiki/Breaking-Changes) has been updated in case of breaking changes.

### CODE REVIEW

File: src/card.scss [hunk 1]
```diff
@@ -41,6 +42,13 @@ $fd-card-content-avatar-text-spacing: 0.5rem !default;
     @include fd-reset();
     @include fd-flex();
 
+    padding: 1rem;
+    background: $fd-card-default-body-background-color;
+    border-bottom: $fd-card-header-border;
+    border-radius: $fd-card-header-border-radius $fd-card-header-border-radius 0 0;
+    text-decoration: none;
+    cursor: pointer;
+
```
Comment (InnaAtanasova) [inline src/card.scss:51]: why did you move these rules here?
not that it's a big deal but usually all @include statements are at the beginning (except rtl) and after that come the normal css rules

Comment (JKMarkowski) [reply src/card.scss:51]: It's done because it is totally `fd-hover/active/selected` are totally different mixins than `flex/reset`. I treat them same as `&:hover/active/selected`. That's why those properties have been moved up

**Review by mikerodonnell89 — APPROVED** [review]

**Review by InnaAtanasova — APPROVED** [review]

File: src/card.scss (hunk 1 above)
Comment (InnaAtanasova) [reply src/card.scss:51]: of course they are different. But the order of the css should not be important :) it's just a common practice in scss world that the include statements are at the beginning.
